src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx

```diff
@@ -92,18 +92,37 @@ import {getOrgCreationAllowancesThunk} from 'src/identity/allowances/actions/thu
 import {generateProviderMap} from 'src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/utils/generateProviderMap'
 import {SafeBlankLink} from 'src/utils/SafeBlankLink'
 
+// Eventing
+import {event} from 'src/cloud/utils/reporting'
+import {
+  CreateOrgOverlayEvent,
+  multiOrgTag,
+} from 'src/identity/events/multiOrgEvents'
+
 const switchToNewOrgButtonStyle = {
   color: 'black',
   textDecoration: 'underline',
   cursor: 'pointer',
 }
 
-const SwitchToNewOrgButton = (url: string): JSX.Element => {
+const SwitchToNewOrgButton = (
+  url: string,
+  newOrg: CreatedOrg,
+  currentOrgId
```

Comment:
Let's talk about method design as it relates to method dependencies and interfaces.

Currently / before this change, the `SwitchToNewOrgButton` takes in a `url`, which makes sense - this component's main purpose is to render a button that navigates to a link, so being able to set the destination url of that link is completely reasonable.

The change we want is to be able to fire an event when the button is clicked. We have two options here - we can pass in all the things that the event needs to be able to fire correctly, which is the route you're currently taking. The event needs the current org id and the org that is being switched to. If we pass those in to this component, suddenly this component now has many responsibilities - it needs to be aware of not just the current and new org, but these concepts in general. Whereas this component used to be kind of dumb and only needed a url, it now needs system-level concepts - orgs.

This will work, but it leaks information about the system into a really dumb component, which causes two issues: it makes the component much more difficult to reuse (it can only be reused if the thing using it has access to current and new orgs), and it makes the intent of the component a little muddled.

An alternative to passing in current and new org is to pass in a callback function that is called when the button is clicked. So rather than pass in `newOrg` and `currentOrgId` to `SwitchToNewOrgButton`, if this code instead passed in `handleEventing`, which was constructed outside of this method, then `handleEventing` would abstract its dependencies on `newOrg` and `currentOrgId` away from the `SwitchToNewOrgButton`. This sidesteps the two issues mentioned above - now only `handleEventing` needs to worry about new and currente orgs, and `SwitchToNewOrgButton` is much more reusable, since a callback to be called `onClick` is much more generic, and can be a no-op.

Here's something along the lines that I was thinking:

```ts
const SwitchToNewOrgButton = (
  url: string,
  handleClick? = () => {}
) => {
  return (
    <a
      href={url}
      data-testid="go-to-new-org--link"
      style={switchToNewOrgButtonStyle}
      onClick={handleClick}
}
```

then when it's called:

```ts
const handleEventing = () => {
  event(CreateOrgOverlayEvent.SwitchToNewOrg, multiOrgTag, {
    currentOrgId,
    createdOrgId: newOrg.id,
    createdOrgName: newOrg.name,
  })
}
const switchToOrgLink =
  newOrg.provisioningStatus === 'provisioned'
    ? () => SwitchToNewOrgButton(newOrgUrl, handleEventing)
    : null
```

Now in the grand scheme of things, this button probably won't be reused very much, since it's defined in a very narrow context. But these principles are applicable to the design of methods, classes, components, etc.

Reply:
your reasoning makes perfect sense. It's a simple change but it does a good job clearing up the blurriness of "labour division" that currently exists. Great suggestion. I'll update accordingly